test/cql-pytest/test_secondary_index.py

```diff
@@ -1792,3 +1792,167 @@ def test_unbuilt_index_not_used(cql, test_keyspace, cassandra_bug):
             assert "ALLOW FILTERING" in str(e)
+
+# Utility function waiting (up to a timeout) for the given index to be built.
+# Uses the "IndexInfo" system table, supported by both Scylla and Cassandra.
+def wait_for_index(cql, keyspace, index_name, timeout_sec=60):
+    start_time = time.time()
+    while time.time() < start_time + timeout_sec:
+        if list(cql.execute(f"SELECT index_name FROM system.\"IndexInfo\" WHERE table_name = '{keyspace}' and index_name = '{index_name}'")):
+            return
+        time.sleep(0.1)
+    pytest.fail(f"Timeout ({timeout_sec} seconds) waiting for index {keyspace}.{index_name}")
+
+# The following test starts a filtering request (with ALLOW FILTERING),
+# pages through the results, and between two pages adds a secondary-index
+# that could be used - or not - for continuing the request, but certainly
+# shouldn't break the request. Reproduces #18992.
+@pytest.mark.xfail(reason="issue #18992")
+def test_paging_and_create_index(cql, test_keyspace):
+    count = 20
+    with new_test_table(cql, test_keyspace,
+            "p int, v int, PRIMARY KEY (p)") as table:
+        stmt = cql.prepare(f"INSERT INTO {table} (p, v) VALUES (?, ?)")
+        # Add 'count' partitions with v=17 in all of them, so the filter
+        # "WHERE v=17" will return all rows.
+        for i in range(count):
+            cql.execute(stmt, [i, 17])
+        page_size = 7
+        stmt = SimpleStatement(f"SELECT p FROM {table} WHERE v=17 ALLOW FILTERING", fetch_size=page_size)
+        # Save the result of a full query (this would be count rows ordered
+        # by token(p) order) so we can later compare this to what we get
+        # when we insert a CREATE INDEX between pages.
+        expected = list(cql.execute(stmt))
+        # Run the same paged query again but this time use the page-by-page
+        # API, and stick a CREATE INDEX ON v between the first and second page.
+        got = []
+        r = cql.execute(stmt)
+        assert len(r.current_rows) == page_size  # sanity check
+        got.extend(r.current_rows)
+        index_name = unique_name()
+        cql.execute(f"CREATE INDEX {index_name} ON {table}(v)")
+        # We wait for the index to be built, since we don't want to reproduce
+        # #7963 again here (an index getting used before actually built).
+        wait_for_index(cql, test_keyspace, index_name)
+        while r.has_more_pages:
+            r = cql.execute(stmt, paging_state=r.paging_state)
+            assert len(r.current_rows) <= page_size # sanity check
+            got.extend(r.current_rows)
+        assert expected == got
+
+# This test is the same as the previous one, but in this test the request
+# filters on both v1 and v2 is already using an index for column v2, and
+# now between the pages we add an index for v1. Reproduces #18992.
+@pytest.mark.xfail(reason="issue #18992")
+def test_paging_and_create_index2(cql, test_keyspace):
```

Comment:
Maybe there's a better name for this function?

Reply:
Do you have a good suggestion?
As I explained in the detailed comment, this test "is the same as the previous one, but...". I didn't want to have a 100-character function name explaining this "but" so I just used "2" :-)  I honestly think that a comment serves the readers better than a 100-character function name.